### PR TITLE
Respect also runtime environment when linking packages in Search

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/adviser/issues/2223

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: thoth/adviser/wraps/thoth_search_package.py [hunk 1]
```diff
@@ -53,12 +56,18 @@ def should_include(cls, builder_context: "PipelineBuilderContext") -> Generator[
...
+        runtime_environment = self.context.project.runtime_environment
         for package_version_tuple in state.resolved_dependencies.values():
             state.justification.append(
                 {
                     "type": "INFO",
                     "link": self._search_package_url.format(
-                        package_name=package_version_tuple[0], package_version=package_version_tuple[1]
+                        package_name=quote(package_version_tuple[0]),
+                        package_version=quote(package_version_tuple[1]),
+                        index_url=quote(package_version_tuple[2]),
```
Comment (Gkrumbach07) [inline thoth/adviser/wraps/thoth_search_package.py:67]: Search UI encodes the index_url. Use `urllib.parse.quote(package_version_tuple[2], safe="")` for a properly formatted url

Comment (fridex) [reply thoth/adviser/wraps/thoth_search_package.py]: Oh, nice! Didn't know this was possible. Adjusted. PTAL 👍🏻

Comment (Gkrumbach07) [reply thoth/adviser/wraps/thoth_search_package.py]: LGTM

**Pull request merged**